### ISSUE TITLE
Windows: FormatException: Missing extension byte (at offset 1)

### ISSUE THREAD

**Issue by kirillvikhlyaev** [llfbandit/record#232]
**Package version**: 5.0.1

**Describe the bug**
It is not possible to start recording on Windows.

**To Reproduce**
1. Go to 'record/example'
2. Run example on Windows
3. Tap on record button
4. Nothing happens. The console recorded the error "flutter: FormatException: Missing extension byte (at offset 1)"

**Desktop (please complete the following information):**
 - OS: Windows 10

**Comment by llfbandit** [comment]
Thanks for the report.
This error should come from UTF conversion.

If you're coming from the example provided, I guess the error is thrown from `listInputDevices`?
In such case, can you print (or screen) your input devices here?

**Comment by PageZhu** [comment]
I have encountered this issue today with `listInputDevices` method. Here is a screenshot of my devices
![image](https://github.com/llfbandit/record/assets/60153829/043572f9-f2e2-4941-b76f-99c924497f63)

**Comment by SirusDoma** [comment]
Same issue here, I debugged through the code and I found out that the platform channel built-in from flutter attempted to decode all of the device friendly name strings with the utf8 decoder and then fail, throwing an exception. [Specifically, this part of code](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/services/message_codecs.dart#L502)

My microphone is `RØDE NT-USB Mini` and I tried to decode the chars one by one and it always failed to decode the `Ø` char. I'm not quite sure how to handle this unrecognized character but think there are a few solutions to fix this:

1. Ignore characters in the friendly name in the native code whenever they cannot be parsed with the UTF-8 decoder. 
2. Provide a list with alternative names in a separate method. There should be more "friendlier names" that don't contain weird character, see image above or below (the alternative names are in English)
3. Provide a custom `MessageCodec` class to handle the string.

(More friendlier: `Desktop Microphone`)
![image](https://github.com/llfbandit/record/assets/9136829/76612c15-f045-4bed-87c5-dbe8feaf2f4c)
